enzyme/test/Integration/ReverseMode/allocatedtape_err.c

```diff
@@ -0,0 +1,21 @@
+// RUN: if [ %llvmver -ge 10 ]; then %clang -std=c11 -g -O0 %s -S -emit-llvm -o -  %loadClangEnzyme -Xclang -verify; fi
+// RUN: if [ %llvmver -ge 10 ]; then %clang -std=c11 -g -O1 %s -S -emit-llvm -o -  %loadClangEnzyme -Xclang -verify; fi
+// RUN: if [ %llvmver -ge 10 ]; then %clang -std=c11 -g -O2 %s -S -emit-llvm -o -  %loadClangEnzyme -Xclang -verify; fi
+// RUN: if [ %llvmver -ge 10 ]; then %clang -std=c11 -g -O3 %s -S -emit-llvm -o -  %loadClangEnzyme -Xclang -verify; fi
+// RUN: if [ %llvmver -ge 12 ]; then %clang -std=c11 -g -O0 %s -S -emit-llvm -o -  %newLoadClangEnzyme -Xclang -verify; fi
+// RUN: if [ %llvmver -ge 12 ]; then %clang -std=c11 -g -O1 %s -S -emit-llvm -o -  %newLoadClangEnzyme -Xclang -verify; fi
+// RUN: if [ %llvmver -ge 12 ]; then %clang -std=c11 -g -O2 %s -S -emit-llvm -o -  %newLoadClangEnzyme -Xclang -verify; fi
+// RUN: if [ %llvmver -ge 12 ]; then %clang -std=c11 -g -O3 %s -S -emit-llvm -o -  %newLoadClangEnzyme -Xclang -verify; fi
+
+extern int enzyme_allocated, enzyme_tape;
+void __enzyme_reverse(void*, ...);
+
+void f(double* x) {
+    x[0] += x[1]*x[1];
+    x[1] = -x[1];
+}
```

Comment:
Looks like for LLVM 10+ the caching infra does a better job and instead emits this error. Mind doing a simpler test case (perhaps x[0] = x[1]*x[1]), and opening an issue to inspect the LLVM 9 cache size.

```
 error: 'error' diagnostics expected but not seen: 
  File /home/runner/work/Enzyme/Enzyme/enzyme/test/Integration/ReverseMode/allocatedtape_err.c Line 19: Enzyme: need 24 bytes have 1 bytes
error: 'error' diagnostics seen but not expected: 
  File /home/runner/work/Enzyme/Enzyme/enzyme/test/Integration/ReverseMode/allocatedtape_err.c Line 19: Enzyme: need 8 bytes have 1 bytes
  ```

Reply:
Hmm - Are you saying that with llvm version  9, you get a 24 byte tape, and with llvm version 10, you get an 8 byte tape? I ran this with llvm 15.0.7 and got a 24 byte tape. Nevertheless, I will try to find a simpler example where the tape size is incontrovertibly 8.

Reply:
In part, I also imagine it depends on optimization level, I think.